src/org/sosy_lab/common/log/MultiLineLogFormatter.java

```diff
@@ -0,0 +1,75 @@
+package org.sosy_lab.common.log;
```

Comment:
License header is missing.

Reply:
- added licence header
- added static factory methods, but no `LoggingOption`s constructor
- Refactored coloured format feature into abstract class

We should maybe come up with a better name for the formatter. I named it `MultilineLogFormatter`, because the first line of the logmessage is displayed inline with:

`timestamp` `level` `message` [`component`:`class`.`method`]   
`additional optional message lines`

Reply:
Hm, that does not really differ from the behavior of the other formatter, right? It also shows all lines of the message.

Is the main difference thus the timestamp? Then the name could be based on this.
Can other differences be eliminated?

Reply:
The main difference is in fact the timestamp, so this could be an option.

The other difference is that we wanted to have the most important information in chronological order on the first line:
- timestamp
- level: if the log message becomes very long, you have to look for the importance  
- message: only the first line, usually contains message, more lines can be seen as additional information, such as the stack trace
- component: since the vcloud doesn't fully leverage that feature yet and due to it's length it was moved to the end.

So the difference to the `FileLogFormatter` is only the position of the component descriptor, but it differs greatly to the `ConsoleLogFormatter`. We use this formatter for both the file and console logging. 
Technically we could refactor `FileLogFormatter` to allow the usage of colours as well and move to it. But then we should rename that one to TimestampedLogFormatter?

Reply:
We could generalize it completely, as well, so that users can provide a formatting string like:
`{timestamp} {level} {message0} ({component}.{method})\n{message1+}`

Reply:
> Technically we could refactor `FileLogFormatter` to allow the usage of colours as well and move to it. But then we should rename that one to TimestampedLogFormatter?

That would be nice. The current `FileLogFormatter` would then stay, but be changed to extend `TimestampedLogFormatter` and marked as `@Deprecated`.

But I note that the current code of `TimestampedLogFormatter` is still different from `FileLogFormatter`. Do you intend to change that?

I would prefer to not invent our own customization mechanism for the message. We do not want to implement our own logging framework.

Reply:
>But I note that the current code of TimestampedLogFormatter is still different from FileLogFormatter. Do you intend to change that?

I guess the vcloud can live with the changed order of log entries and there might even be a incentive to you use the component logging mechanism more.

>That would be nice. The current FileLogFormatter would then stay, but be changed to extend TimestampedLogFormatter and marked as @Deprecated.

Sounds good, I'll make the necessary changes then.